packages/hash/frontend/src/blocks/page/Comments/CommentBlock.tsx

```diff
@@ -294,13 +307,21 @@ export const CommentBlock: FunctionComponent<CommentProps> = ({
           <CommentBlockMenuItem
             title="Delete Comment"
             icon={<FontAwesomeIcon icon={faTrash} />}
-            // @todo Commented implement functionality
             onClick={() => {
+              setDeleteConfirmationDialogOpen(true);
               commentMenuPopupState.close();
             }}
           />
         ) : null}
       </CommentBlockMenu>
+
+      <CommentBlockDeleteConfirmationDialog
+        container={ref.current}
```

Comment:
Reading refs in render isn't safe, because refs aren't updated until post-render, so this will be out of date. If you need access to a ref in render, use state + a callback ref setting the state instead.

Reply:
Changed in ae86e90b48ead5d96ed647aeb7ea007827764059. Thank you.